### PR TITLE
Create Models and Controller

### PR DESCRIPTION
**Pull request by Lemaitin**

### CODE REVIEW

**Comment by Druid-Team-Lead** [comment]
Please add .gitignore file to the project.
Example: https://github.com/github/gitignore/blob/master/VisualStudio.gitignore

File: ToDoApi/ToDoApi/Startup.cs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+		}
+
+		public IConfiguration Configuration { get; }
+
+		// This method gets called by the runtime. Use this method to add services to the container.
+		public void ConfigureServices(IServiceCollection services)
+		{
+
+			services.AddDbContext<TasksContext>(x => x.UseSqlServer("Data Source=MEGATA4KA;Initial Catalog=TaskManager;Integrated Security=True;Connect Timeout=30;"));
+			//services.AddDatabaseDeveloperPageExceptionFilter();
```
Comment (dasokume) [inline ToDoApi/ToDoApi/Startup.cs]: Do we need this?

File: ToDoApi/ToDoApi/Models/TodoItem.cs [hunk 2]
```diff
@@ -0,0 +1,17 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
```
Comment (dasokume) [inline ToDoApi/ToDoApi/Models/TodoItem.cs]: Remove unused **using**

File: ToDoApi/ToDoApi/Models/TasksContext.cs [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+
+namespace ToDoApi.Models
+{
+    public class TasksContext : DbContext
+    {
+        public DbSet<TodoItem> Tasks { get; set; }
```
Comment (dasokume) [inline ToDoApi/ToDoApi/Models/TasksContext.cs]: Add extra empty line here

File: ToDoApi/ToDoApi/Models/TasksContext.cs [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+        public TasksContext(DbContextOptions<TasksContext> options)
+            : base(options)
+        {
+            Database.EnsureCreated();
+        }
```
Comment (dasokume) [inline ToDoApi/ToDoApi/Models/TasksContext.cs]: Here as well

File: ToDoApi/ToDoApi/Controllers/TodoController.cs [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+{
+    [ApiController]
+    [Route("api/[controller]")]
+    public class TodoController : ControllerBase
+    {
+        TasksContext db;
+
+        public TodoController(TasksContext context)
+        {
+            db = context;
```
Comment (dasokume) [inline ToDoApi/ToDoApi/Controllers/TodoController.cs]: Remove from the constructor. Create a new class called DB initializer

File: ToDoApi/ToDoApi/Controllers/TodoController.cs [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+        [HttpGet("{id}")]
+        public async Task<ActionResult<TodoItem>> Get(int id)
+        {
+            TodoItem task = await db.Tasks.FirstOrDefaultAsync(x => x.Id == id);
+            if (task == null)
+                return NotFound();
+            return new ObjectResult(task);
+        }
+
+        // POST api/tasks
```
Comment (dasokume) [inline ToDoApi/ToDoApi/Controllers/TodoController.cs]: Delete comments

File: ToDoApi/ToDoApi/Controllers/TodoController.cs [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+                db.Tasks.Add(new TodoItem { Name = "Work", IsComplete = true });
+                db.Tasks.Add(new TodoItem { Name = "Cooking", IsComplete = false });
+                db.SaveChanges();
+            }
+        }
+        [HttpGet]
+        public async Task<ActionResult<IEnumerable<TodoItem>>> Get()
+        {
+            return await db.Tasks.ToListAsync();
+        }
```
Comment (dasokume) [inline ToDoApi/ToDoApi/Controllers/TodoController.cs]: Add extra empty line

File: ToDoApi/ToDoApi/Controllers/TodoController.cs [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+        [HttpGet("{id}")]
+        public async Task<ActionResult<TodoItem>> Get(int id)
+        {
+            TodoItem task = await db.Tasks.FirstOrDefaultAsync(x => x.Id == id);
```
Comment (dasokume) [inline ToDoApi/ToDoApi/Controllers/TodoController.cs]: Create TodoItemRepository class to work with DB.
Create TodoItemService class to work with the repository.
Use this new service in your controller

**Review by dasokume — CHANGES_REQUESTED** [review]
Please add .gitignore file to the project.
Example: https://github.com/github/gitignore/blob/master/VisualStudio.gitignore

File: ToDoApi/ToDoApi/Startup.cs [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+			Configuration = configuration;
+		}
+
+		public IConfiguration Configuration { get; }
+
+		// This method gets called by the runtime. Use this method to add services to the container.
+		public void ConfigureServices(IServiceCollection services)
+		{
+
+			services.AddDbContext<TasksContext>(x => x.UseSqlServer("Data Source=MEGATA4KA;Initial Catalog=TaskManager;Integrated Security=True;Connect Timeout=30;"));
```
Comment (dasokume) [inline ToDoApi/ToDoApi/Startup.cs]: Move connection string to appsettings.json file

File: ToDoApi/ToDoApi/Context/TodoDbContext.cs [hunk 10]
```diff
@@ -0,0 +1,21 @@
+﻿using Microsoft.EntityFrameworkCore;
+using ToDoApi.Models;
+
+namespace ToDoApi.Context
+{
+	public class TodoDbContext : DbContext
+    {
+        public DbSet<TodoItem> TodoItems { get; set; }
```
Comment (dasokume) [inline ToDoApi/ToDoApi/Context/TodoDbContext.cs:8]: Add empty line

File: ToDoApi/ToDoApi/Context/TodoDbContext.cs [hunk 11]
```diff
@@ -0,0 +1,21 @@
...
+        public TodoDbContext(DbContextOptions<TodoDbContext> options)
+            : base(options)
+        {
+            Database.EnsureCreated();
+        }
+        protected override void OnModelCreating(ModelBuilder modelBuilder)
```
Comment (dasokume) [inline ToDoApi/ToDoApi/Context/TodoDbContext.cs:16]: Add empty line

File: ToDoApi/ToDoApi/Controllers/TodoController.cs [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+using Microsoft.AspNetCore.Mvc;
+using System.Threading.Tasks;
+using ToDoApi.Models;
+using ToDoApi.Service.Interfaces;
+using ToDoApi.Repository.Interfaces;
+
+namespace ToDoApi.Controllers
+{
+	[ApiController]
+    [Route("api/[controller]")]
```
Comment (dasokume) [inline ToDoApi/ToDoApi/Controllers/TodoController.cs:10]: Fix formatting in this file

File: ToDoApi/ToDoApi/Models/TodoItem.cs [hunk 13]
```diff
@@ -0,0 +1,13 @@
+﻿using System.ComponentModel.DataAnnotations;
+
+namespace ToDoApi.Models
+{
+	public class TodoItem
```
Comment (dasokume) [inline ToDoApi/ToDoApi/Models/TodoItem.cs:5]: Fix formatting

File: ToDoApi/ToDoApi/Models/TodoItem.cs [hunk 14]
```diff
@@ -0,0 +1,13 @@
...
+    {
+        [Key]
+        public int Id { get; set; }
```
Comment (dasokume) [inline ToDoApi/ToDoApi/Models/TodoItem.cs:8]: Add empty line between properties

File: ToDoApi/ToDoApi/Repository/Interfaces/ITodoItemRepository.cs [hunk 15]
```diff
@@ -0,0 +1,18 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
```
Comment (dasokume) [inline ToDoApi/ToDoApi/Repository/Interfaces/ITodoItemRepository.cs]: Remove unused using

File: ToDoApi/ToDoApi/Repository/Interfaces/ITodoItemRepository.cs [hunk 16]
```diff
@@ -0,0 +1,18 @@
...
+using System.Threading.Tasks;
+using ToDoApi.Models;
+
+namespace ToDoApi.Repository.Interfaces
+{
+    public interface ITodoItemRepository : IDisposable
+    {
+        Task<IEnumerable<TodoItem>> GetItemList();
```
Comment (dasokume) [inline ToDoApi/ToDoApi/Repository/Interfaces/ITodoItemRepository.cs:10]: Add empty line between methods

File: ToDoApi/ToDoApi/Service/Interfaces/ITodoService.cs [hunk 17]
```diff
@@ -0,0 +1,14 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
```
Comment (dasokume) [inline ToDoApi/ToDoApi/Service/Interfaces/ITodoService.cs]: Remove unused using

File: ToDoApi/ToDoApi/Service/Interfaces/ITodoService.cs [hunk 18]
```diff
@@ -0,0 +1,14 @@
...
+using System.Threading.Tasks;
+using ToDoApi.Models;
+
+namespace ToDoApi.Service.Interfaces
+{
+	public interface ITodoService
+	{
+		Task<IEnumerable<TodoItem>> GetItemList();
```
Comment (dasokume) [inline ToDoApi/ToDoApi/Service/Interfaces/ITodoService.cs:9]: Add empty line between methods

File: ToDoApi/ToDoApi/Service/TodoService.cs [hunk 19]
```diff
@@ -0,0 +1,29 @@
...
+namespace ToDoApi.Service
+{
+	public class TodoService : ITodoService
+	{
+		private ITodoItemRepository _itodoItemRepository;
+
+		public TodoService(ITodoItemRepository todoItemRepository)
+		{
+			_itodoItemRepository = todoItemRepository;
+		}
```
Comment (dasokume) [inline ToDoApi/ToDoApi/Service/TodoService.cs:18]: Add empty line

File: ToDoApi/ToDoApi/Startup.cs [hunk 20]
```diff
@@ -0,0 +1,60 @@
...
+	public class Startup
+	{
+		public Startup(IConfiguration configuration)
+		{
+			Configuration = configuration;
+		}
+
+		public IConfiguration Configuration { get; }
+
+		// This method gets called by the runtime. Use this method to add services to the container.
```
Comment (dasokume) [inline ToDoApi/ToDoApi/Startup.cs]: Remove trash comments in file

**Review by dasokume — CHANGES_REQUESTED** [review]

File: ToDoApi/ToDoApi/Controllers/TodoController.cs [hunk 21]
```diff
@@ -0,0 +1,38 @@
...
+		public async Task<ActionResult<IEnumerable<TodoItem>>> Get()
+		{
+			return Ok (await _todoService.GetItemList());
+		}
+
+		[HttpGet("{id}")]
+		public async Task<ActionResult<TodoItem>> Get(int id)
+		{
+			TodoItem todoItem = await _todoService.GetItem(id);
+			if (todoItem == null)
```
Comment (dasokume) [inline ToDoApi/ToDoApi/Controllers/TodoController.cs:31]: Empty line after service call and before return statement

**Review by dasokume — CHANGES_REQUESTED** [review]

**Comment by dasokume** [comment]
Well done!

**Review by dasokume — APPROVED** [review]

**Pull request merged**